### PR TITLE
Adding instructions for the MC prototype to be tested end to end over Azure Policy

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/agents/gc/README.md [hunk 1]
```diff
@@ -53,4 +53,146 @@ sudo pwsh
...
+
+Go to Azure Portal | Azure Arc | Add your infrastructure for free | Add your existing server | Generate script. Specify the desired Azure subscription, resource group and OS (Linux) for the device. Copy and run the generated instructions in the local shell on the device to be onboarded.
+
+## 6. Uploading the ZIP artifacts package
+
+The generated artifacts ZIP package OSConfig_Proto_Policy.zip needs to be uploaded to Azure.
+
+Go to Azure Portal and create a Storage Account. In there, go to Containers and create a new container. Then inside of that container upload the artifacts ZIP package OSConfig_Proto_Policy.zip.
+
+Important: after each update of the ZIP package the policy definition will need to be updated.
```
Comment (AhmedBM) [inline src/agents/gc/README.md:68]: Can we use the `Important` blockquote?
See https://github.com/community/community/discussions/16925 for more info

> **Important**
> This is important

File: src/agents/gc/README.md [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
 
-## About Machine Configuration
+## 1. About Machine Configuration
 
 Read about Azure Automanage Machine Configuration (formerly called Azure Policy Guest Configuration) at [learn.microsoft.com/machine-configuration](https://learn.microsoft.com/en-us/azure/governance/machine-configuration/).
 
-## Regenerating the NRP code for a changed resource class:
+## 2. Regenerating the NRP code for a changed resource class
 
 To regenerate code, see [codegen.cmd](codegen.cmd). Warning: regenerating code will overwrite all customizations and additions specific to OSConfig.
```
Comment (AhmedBM) [inline src/agents/gc/README.md:9]: Can we use the warning blockquote?
See https://github.com/orgs/community/discussions/16925 for more info

> **Warning**
> This is a warning

**Review by AhmedBM — DISMISSED** [review]
Looks good, just a few minor suggestions

File: src/agents/gc/README.md [hunk 3]
```diff
@@ -53,4 +53,146 @@ sudo pwsh
 $x = Get-GuestConfigurationPackageComplianceStatus -path <path to the ZIP>/OSConfig_Proto_Policy.zip -Verbose
 $x.resources[0].properties
 ```
-In addition to the MC traces written to the the PowerShell console, you can also see the NRP's own log at `/var/log/osconfig_gc_nrp.log`.
\ No newline at end of file
+In addition to the MC traces written to the the PowerShell console, you can also see the NRP's own log at `/var/log/osconfig_gc_nrp.log`.
+
+## 5. Onboarding a device to Arc
+
+Go to Azure Portal | Azure Arc | Add your infrastructure for free | Add your existing server | Generate script. Specify the desired Azure subscription, resource group and OS (Linux) for the device. Copy and run the generated instructions in the local shell on the device to be onboarded.
```
Comment (simonjaeger) [inline src/agents/gc/README.md:60]: Instructions are clear, but may be easier to read as a list? For example:

1. Go to Azure Portal
2. Select Azure Arc
3. ...

**Review by simonjaeger — APPROVED** [review]
:shipit:

**Review by AhmedBM — APPROVED** [review]
👌

**Pull request merged**